x/gov/client/utils/query.go

```diff
@@ -149,7 +154,9 @@ func QueryDepositByTxQuery(
 		fmt.Sprintf("%s='%s'", tags.Depositor, []byte(params.Depositor.String())),
 	}
 
-	infos, err := tx.SearchTxs(cliCtx, cdc, tags)
+	// NOTE: SearchTxs is used to facilitate the txs query which does not currently
+	// support configurable pagination.
+	infos, err := tx.SearchTxs(cliCtx, cdc, tags, 1, 30)
```

Comment:
IMHO a paginated response should be structured differently, e.g.:

```
{
  "items": [ ... ]
  "count": 10 // count of actual items in this response
  "page": ... // it should mirror user's value
  "total": ... // total number of items that match the criteria, including the returned ones  
}
```
 The `total` field is particularly important for users as it tells them how to query for next items.

Reply:
Agree here

Reply:
> The `total` field is particularly important for users as it tells them how to query for next items.

This is a valid point. But IMHO this involves tendermint changes because its tendermint side knows how many in total the query returned. How about let me implement this in another PR to save conflict resolving time?

By the way, for our use case (query txs with tag=tx.height=42), we firstly query host:8080/blocks/42, it already returned `numTx` in block header of this block, so we always know how much in total

Reply:
Please file a releveant issue against Tendermint then

Reply:
Has anyone opened a relevant Tendermint issue for this?

Reply:
> Has anyone opened a relevant Tendermint issue for this?

I created one https://github.com/tendermint/tendermint/issues/3113